src/components/Button/Button.tsx

```diff
@@ -44,16 +47,26 @@ export const Button = ({
   const fullStyles = isFull ? 'w-full' : '';
   const centerStyles = isCenter ? 'relative left-1/2 -translate-x-1/2 ' : '';
   const disabledStyles = isDisabled ? styles.disabled : '';
+  const iconSize = size === 'lg' ? '6' : '4';
+  // const forDisabled = isDisabled ? styles['disabled'] : '';
+  const stylesType = isDisabled ? styles['disabled'] : styles[isType];
 
   return (
     <button
       type="button"
       aria-label={title ? title : 'Icon button'}
-      className={`${styles.core} ${styles[isType]} ${buttonSizes[size]} ${shadowStyles} ${fullStyles} ${centerStyles} ${disabledStyles}`}
+      className={`${styles.core} ${stylesType} ${buttonSizes[size]} ${shadowStyles} ${fullStyles} ${centerStyles} ${disabledStyles} `}
       disabled={isDisabled}
       {...rest}
     >
-      <span className={`${title && icon ? 'mr-2' : ''} `}>{icon}</span>
+      {icon && (
```

Comment:
Instead of passing className we should look at adding `gap-0.5` on `styles.core` and see if that works. Allowing className would open up the door for any styles to be passed down, something we should avoid. I did this on all my first components and realized it was probably a bad thing.

Reply:
Just quickly tested this and it will work, but you'll wanna use `gap-2`